app/components/connectionForm.js

```diff
@@ -1,36 +1,27 @@
-import React, {Component, PropTypes} from 'react';
+import React, {PropTypes} from 'react';
 
-export default class Connect extends Component {
-  constructor(props) {
-    super(props);
-    this.handleConnect = this.handleConnect.bind(this);
-  }
-  handleConnect() {
-    const ip = this.refs.ip.value;
-    this.props.connect(ip);
-  }
-  renderConnect() {
-    return (
-      <div>
-        <input ref="ip" id="ip" name="ip" placeholder="Ip Eg: 12.0.0.1/localhost"/>
-        <button type="button" className="btn btn-success" onClick={this.handleConnect}>Connect</button>
-      </div>
-    );
-  }
-  renderDisconnect() {
-    return (
-      <div>
-        <button type="button" className="btn btn-danger" onClick={this.props.handleDisconnect}>Disconnect</button>
-      </div>
-    );
-  }
-  render() {
-    return this.props.status === 'CONNECTED' ? this.renderDisconnect() : this.renderConnect();
-  }
-}
+const ConnectButton = props => {
+  return <button type="button" className="btn btn-success" onClick={props.handleConnect}>Connect</button>;
+};
 
-Connect.propTypes = {
-  status: PropTypes.string,
-  connect: PropTypes.func,
+ConnectButton.propTypes = {
+  handleConnect: PropTypes.func
+};
+
+const DisconnectButton = props => {
+  return <button type="button" className="btn btn-danger" onClick={props.handleDisconnect}>Disconnect</button>;
+};
+
+DisconnectButton.propTypes = {
   handleDisconnect: PropTypes.func
 };
+
+const Connect = props => {
+  return props.status === 'CONNECTED' ? <DisconnectButton {...props}/> : <ConnectButton {...props}/>;
+};
+
+Connect.propTypes = {
+  status: PropTypes.string
+};
+
+export default Connect;
```

Comment:
Not sure what the convention is, is it a good practice to have stateless components within another component or should they be in separate files? On the other hand, these are most likely not going to be re used anywhere.